File: testing/php/testFilter.php
<?php
 
require_once '/Users/christopher/Code/sl-php/SoftLayer/SoapClient.class.php';
 
 

$apiUser = '';
$key = '';


$startDate = new DateTime('2014-11-01T13:05:25-06:00');
$endDate = new DateTime('2014-12-01T09:53:51-06:00');
 
$accountClient = SoftLayer_SoapClient::getClient('SoftLayer_Account', '391780', $apiUser, $key);

$accountFilter = new SoftLayer_ObjectMask();
$accountFilter->id = '391780';
$accountClient->addHeader('SoftLayer_AccountInitParameters',$accountFilter);

$filter = new stdClass();
$filter->invoices = new stdClass();
// $filter->invoices->accountId = new stdClass();
// $filter->invoices->accountId->operation = '= 391780';
$filter->invoices->createDate = new stdClass();
$filter->invoices->createDate->operation = 'betweenDate';
$filter->invoices->createDate->options = array();
$filter->invoices->createDate->options[0] = new stdClass();
$filter->invoices->createDate->options[0]->name = 'startDate';
$filter->invoices->createDate->options[0]->value = array($startDate->format('m/d/Y H:i:s'));
$filter->invoices->createDate->options[1] = new stdClass();
$filter->invoices->createDate->options[1]->name = 'endDate';
$filter->invoices->createDate->options[1]->value = array($endDate->format('m/d/Y H:i:s'));

$objectMask = new SoftLayer_ObjectMask();
$objectMask->invoices; 
 
$accountClient->setObjectFilter($filter);
$accountClient->setObjectMask($objectMask);

$invoices = $accountClient->getObject();

print_r($invoices);

File: testing/python/testFilter.py
import SoftLayer
from SoftLayer import utils
import sys
import pprint




pp = pprint.PrettyPrinter(indent=4)
client = SoftLayer.Client(username=apiUsername,api_key=apiKey)


items = ['invoices']
theMask  = "mask[%s]" % ','.join(items)
theFilter = {
        'invoices': {
            'createDate': {
                'operation': 'betweenDate',
                'options': [
                    {'name': 'startDate', 'value': ['2014-11-01T09:30:53-07:00']},
                    {'name': 'endDate', 'value': ['2014-12-01T09:30:53-07:00']}
                ]
            }
        }
    }

result = client['SoftLayer_Account'].getInvoices( filter=theFilter);
pp.pprint(result)
